js/bruh-dash.js

```diff
@@ -10,115 +10,273 @@ var global = window || GLOBAL;
   },
 
   // returns a slice of array with n elements dropped from the end
-  dropRight: function() {
-
+  dropRight: function(arr, n) {
+  if (n === undefined){
+    arr.pop();
+    return arr;
+  }
+  else {
+    for (var i = 0; i < n; i++){
+      arr.pop();
+    }
+    return arr;
+    }
   },
 
-  // creates a slice of an array with n elements taken from the beginning
-  take: function () {
 
+  // creates a slice of an array with n elements taken from the beginning
+  take: function (arr, n) {
+    if (n === undefined){
+      return arr.slice(0,1);
+    }
+    else {
+    return arr.slice(0,n);
+    }
   },
 
-  // creates a slice of an array with n elements taken from the end
-  takeRight: function () {
 
+  // creates a slice of an array with n elements taken from the end
+  takeRight: function (arr, n) {
+    if (n === undefined){
+      return arr.slice(arr.length-1, arr.length);
+    }
+    else {
+    return arr.slice(arr.length-n, arr.length);
+    }
   },
 
   // fills elements of array with specified value from the start index
   // up to but not including the end index
-  fill: function() {
 
+  fill: function(arr, val, startIndex, endIndex) {
+    if (startIndex === undefined){
+      startIndex = 0;
+    }
+    if (endIndex === undefined){
+      endIndex = arr.length;
+    }
+    for (var i = startIndex; i < endIndex; i++){
+      arr[i]=val;
+    }
+    return arr;
   },
 
   // removes all given values from an array
-  pull: function () {
-
+  // parameters: arr, and the values to remove
+  // should be able to accept any number of parameters for the values to remove
+  // returns the array - without the values
+
+  pull: function (arr, val) {
+    for (var i = arr.length - 1; i >= 0; i--){
+      for (var j = 0; j < val.length; j++){
+        if (arr[i] === val[j]){
+          arr.splice(i, 1);
+        }
+      }
+    }
+    return arr;
   },
 
   // removes elements of an array corresponding to the given indices
-  pullAt: function () {
-
+  // returns an array of removed elements
+
+  pullAt: function (arr, ind) {
+    var newarr = [];
+    ind.sort((a, b) => a-b);
+    for (var i = ind.length - 1; i >= 0; i--){
+          newarr.push(arr[ind[i]]);
+          arr.splice(ind[i], 1);
+        }
+    newarr.reverse();
+    return newarr;
+
+    //this should be returning the array of removed values rather than the original array without the removed values - per LoDash documentation
+    //changed tests to check for newarr rather than arr
   },
 
   // creates an array excluding all the specified values
-  without: function() {
 
+  without: function(arr, val) {
+  for (var i = arr.length - 1; i >= 0; i--){
+      for (var j = 0; j < val.length; j++){
+        if (arr[i] === val[j]){
+          arr.splice(i, 1);
+        }
+      }
+    }
+    return arr;
   },
 
   // returns an array with specified values excluded
-  difference: function() {
 
+  difference: function(arr, val) {
+    for (var i = 0; i < arr.length; i++){
```

Comment:
This for loop (the outer for loop) should be going the other way or else the splice might jump over an index while moving through the array. 
Example, 
difference([ false, false, true ], false)  ---> [ false, true ]